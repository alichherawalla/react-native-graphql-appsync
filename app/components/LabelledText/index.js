/**
 *
 * LabelledText
 * This component consists of a Label above the text and a value associated
 * to that particular label.
 * Doesn't render anything if value is absent.
 *
 */

import React from 'react'
import PropTypes from 'prop-types'
import { View, StyleSheet } from 'react-native'
import T from 'app/components/T'
import If from 'app/components/If'

function LabelledText({ labelId = '', value = null }) {
  return (
    <If condition={!!value}>
      <View testID="labelled-text" style={styles.labelledView}>
        <T id={labelId} size="small" />
        <T text={value} size="small" />
      </View>
    </If>
  )
}
const styles = StyleSheet.create({
  labelledView: {
    marginVertical: 4,
    marginHorizontal: 4
  }
})
LabelledText.propTypes = {
  labelId: PropTypes.string,
  value: PropTypes.string
}

export default LabelledText
