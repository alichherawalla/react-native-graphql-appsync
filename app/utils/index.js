import { fonts } from 'app/themes'

export const getRandomKey = () =>
  Math.random()
    .toString(36)
    .substr(2, 3)
export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms))

export const rnFontSizeAdapter = type => ({
  fontSize: parseInt(
    fonts.size[type]()[0]
      .replace('font-size: ', '')
      .replace('px;', '')
      .trim(),
    10
  )
})

export const rnFontWeightAdapter = type => ({
  fontSize: parseInt(
    fonts.weights[type]()[0]
      .replace('font-weight: ', '')
      .trim(),
    10
  )
})

export const NEW_EMPLOYEE = {
  employee: {
    address: [{}],
    skills: [{}]
  }
}

export const SIGNUP_CONFIG = {
  header: 'Sign up to buildops',
  hideAllDefaults: true,
  defaultCountryCode: '1',
  signUpFields: [
    {
      label: 'User name',
      key: 'username',
      required: true,
      displayOrder: 1,
      type: 'string'
    },
    {
      label: 'Password',
      key: 'password',
      required: true,
      displayOrder: 2,
      type: 'password'
    },
    {
      label: 'Email',
      key: 'email',
      required: true,
      displayOrder: 4,
      type: 'string'
    }
  ]
}
