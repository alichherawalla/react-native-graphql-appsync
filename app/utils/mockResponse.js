export const employeeData = [
  {
    id: 1,
    firstname: 'John',
    lastname: 'Doe',
    address: [
      {
        line1: '3014 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3020 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 1, name: 'Technician' },
      { id: 2, name: 'Plumber' },
      { id: 3, name: 'Electrician' }
    ]
  },
  {
    id: 2,
    firstname: 'Jonathan',
    lastname: 'Doe',
    address: [
      {
        line1: '3022 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3024 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3026 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3028 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [{ id: 6, name: 'Senior Electrician' }]
  },
  {
    id: 3,
    firstname: 'Steve',
    lastname: 'Doe',
    address: [
      {
        line1: '3028 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 4, name: 'Senior Technician' },
      { id: 5, name: 'Senior Plumber' }
    ]
  },
  {
    id: 4,
    firstname: 'Nathan',
    lastname: 'Doe',
    address: [
      {
        line1: '3022 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3024 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3026 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 4, name: 'Senior Technician' },
      { id: 5, name: 'Senior Plumber' },
      { id: 6, name: 'Senior Electrician' }
    ]
  },
  {
    id: 5,
    firstname: 'Albert',
    lastname: 'Doe',
    address: [
      {
        line1: '3022 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3024 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3026 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      },
      {
        line1: '3028 Dauphine',
        line2: 'St Ste A',
        city: 'New Orleans',
        state: 'LA',
        zipcode: '70117'
      }
    ],
    skills: [
      { id: 4, name: 'Senior Technician' },
      { id: 5, name: 'Senior Plumber' },
      { id: 6, name: 'Senior Electrician' }
    ]
  }
]
