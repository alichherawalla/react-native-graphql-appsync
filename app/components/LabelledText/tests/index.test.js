/**
 *
 * Tests for LabelledText
 *
 */

import React from 'react'
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from 'app/utils/testUtils'
import LabelledText from '../index'

describe('<LabelledText />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(
      <LabelledText value="something" labelId="employee_details" />
    )
    expect(baseElement).toMatchSnapshot()
  })

  it('should contain 1 LabelledText component', () => {
    const { getAllByTestId } = renderWithIntl(
      <LabelledText value="something" labelId="employee_details" />
    )
    expect(getAllByTestId('labelled-text').length).toBe(1)
  })
})
