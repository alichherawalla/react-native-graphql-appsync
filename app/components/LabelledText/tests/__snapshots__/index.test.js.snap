// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LabelledText /> should render and match the snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "marginHorizontal": 4,
          "marginVertical": 4,
        }
      }
      testID="labelled-text"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Object {
              "color": undefined,
              "fontSize": 14,
              "fontWeight": "normal",
            },
          ]
        }
        testID="t"
      >
        Employee Details
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
              "textAlign": "left",
            },
            Object {
              "color": undefined,
              "fontSize": 14,
              "fontWeight": "normal",
            },
          ]
        }
        testID="t"
      >
        something
      </Text>
    </View>
  </View>
</View>
`;
